### ISSUE TITLE
Ctrl+Tab/Ctrl+Shift+Tab should switch tabs

### ISSUE THREAD

**Issue by Sbte** [emesene/emesene#116]
One should be able to switch tabs using Ctrl+Tab (forward) and Ctrl+Shift+Tab (backward), like in emesene 1 or in for example any web browser.

**Comment by DragaO** [comment]
i never used emesene in tabs, but, ctrl+pageup or ctrl+pagedown works well.

by the way, at least in my firefox, ctrl+t doesn't work (opens a new tab) and ctrl+shit+t opens the last closed tab

**Comment by c10ud** [comment]
same as dragao here, tried in emesene1 as well and doesn't work like this

**Comment by Malabarba** [comment]
Sbte, both in firefox and in gnome-terminal the keys you suggested deal with opening tabs. Emesene should either do like firefox (ctrl+Tab and ctrl+shift+Tab)  or like the terminal (ctrl+pageup/pagedown).

**Comment by Sbte** [comment]
Oh, I meant tab indeed. Not t. I'll fix that. Thanks for the fast replies ;)

**Comment by GVisconti84** [comment]
Chromium browser behaves the same as Firefox. I think that CTRL+T and CTRL+SHIFT+T would be really counterintuitive.

**Comment by Malabarba** [comment]
Now that the title was fixed: I'm not sure if I'd prefer ctrl+tab or ctrl+pagedown as default (I'm used to using both), but it'd sure be nice to have an option for that.

**Comment by GVisconti84** [comment]
Probably in the future it would be nice to have customizable shortcut, but now it's not time to think about this :)

**Comment by Sbte** [comment]
According to the code it should work (I think)

```
def on_key_press(self, widget, event):
    '''callback called when a key is pressed on the window'''
    if (event.state & gtk.gdk.CONTROL_MASK) and event.keyval in [
            gtk.keysyms.Tab, gtk.keysyms.ISO_Left_Tab]:
        self.cycle_tabs()
        return True
```

Where page down and page up have

```
def on_key_cycle_tabs(self, accelGroup, window, keyval, modifier):
    '''Catches events like Ctrl+AvPag and consequently changes current
    tab'''

    if not modifier == gtk.gdk.CONTROL_MASK:
        return


    if keyval == gtk.keysyms.Page_Down:
        self.cycle_tabs()
    elif keyval == gtk.keysyms.Page_Up:
        self.cycle_tabs(True)

    return True
```

I tried to make a patch for this, but what happens is that as soon as you press Ctrl+Tab, the focus is on the toolbar instead of on the window itself. This is why the Tab isn't captured by either of the on_key_cycle_tabs or on_key_press functions. This even after modifying the on_key_cycle_tabs function (and _set_accels function) to also handle the Ctrl+Tab. Not sure how to proceed from here...

**Comment by Sbte** [comment]
onTextviewKeyPress(self, widget, event)

from emesene1 is probably the function responsible for this working in emesene1 and not in emesene2

**Comment by marianoguerra** [comment]
I think ctrl+tab is handled by gtk to cycle through widgets that can have focus, that's why we don't support it (I remember that from a discussion a long time ago), I think there is a hack to make it work but I'm not sure.

I think we should keep supporting the current shortcut (and we should document the shortcuts ;)

**Comment by marianoguerra** [comment]
cycle through tabs with ctrl+tab and ctrl+shift+tab, closed by ec4a92721bfaeb6822be68bb30c838f129ae31ed, note:

this doesn't focus the input widget again because you would have to do
the key combo twice for each tab, but you have to focus the input box
manually
